internal/status/status.go

```diff
@@ -22,13 +22,40 @@ var immediateGSInTerminalStatusLock = sync.Mutex{}
 // If an error occurs reporting the status, it will be logged and returned.
 //
 // This function is used by default for reporting status to the local file system unless a different method is specified.
-func ReportStatusToLocalFile(ctx *log.Context, hEnv types.HandlerEnvironment, metadata types.RCMetadata, statusType types.StatusType, c types.Cmd, msg string) error {
+func ReportStatusToLocalFile(ctx *log.Context, hEnv types.HandlerEnvironment, metadata types.RCMetadata, statusType types.StatusType, c types.Cmd, msg string, exitcode ...int) error {
```

Comment:
Is the `exitcode ...int` a hack around go not allowing polymorphism of functions?